contracts/base/KyberSwapRole.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+abstract contract KyberSwapRole {
+  address public owner;
+  bool public isLogicEnabled;
+  mapping(address => bool) public operators;
+  mapping(address => bool) public guardians;
+
+  event TransferOwner(address indexed newOwner);
+  event UpdateOperator(address indexed operator, bool grantOrRevoke);
+  event UpdateGuardian(address indexed guardian, bool grantOrRevoke);
+  event UpdateLogic(bool isEnbaled);
+
+  modifier onlyOwner() {
+    require(msg.sender == owner, 'KyberSwapRole: not owner');
+    _;
+  }
+
+  modifier onlyOperator() {
+    require(operators[msg.sender], 'KyberSwapRole: not operator');
+    _;
+  }
+
+  modifier onlyGuardian() {
+    require(guardians[msg.sender], 'KyberSwapRole: not guardian');
+    _;
+  }
+
+  modifier onlyEnabled() {
```

Comment:
I think we should just name it like `paused` or `unpaused`, better understand

Reply:
yeah paused or unpaused is also ok